app/src/main/java/cm/aptoide/pt/wallet/WalletInstallActivity.kt

```diff
@@ -41,11 +45,59 @@ class WalletInstallActivity : ActivityView(), WalletInstallView {
                                           walletApp: WalletApp) {
     progressView.visibility = View.GONE
     walletInstallSuccessViewGroup.visibility = View.GONE
-    walletInstallViewGroup.visibility = View.VISIBLE
+    closeButton.visibility = View.GONE
 
     appIcon?.let {
       ImageLoader.with(this).load(appIcon, appIconImageView)
     }
+    val downloadModel = walletApp.downloadModel
+    Logger.getInstance()
+        .d("WalletInstallActivity", "download state is " + downloadModel!!.downloadState)
+    if (downloadModel!!.isDownloading) {
+      setDownloadProgress(downloadModel)
+    } else {
+      wallet_install_download_view.visibility = View.GONE
+      progressView.visibility = View.GONE
+      walletInstallViewGroup.visibility = View.VISIBLE
+    }
+  }
```

Comment:
didn't need the verification. Probably some changes made latter on in the code led to not being necessary to verify null

Reply:
It's true that the downloadModel here will not be null (because in the presenter/walletmanager it seems like it always creates a DownloadModel). But that's a problem probably in my part (since I did it), I should've probably flagged it as non-nullable.

However, it's a more general problem. Since it could be null (according to the WalletApp POJO), we should still verify it and not force it. If for some reason the downloadModel is allowed to be null in the future, this could crash. It's just safer and it's the recommended way.